File: option/src/app/python_opensearch/src/rest.py
import os 
from app import app
from flask import jsonify
import requests
import traceback

@app.route('/dept')
def dept():
    try:
        url = "https://"+os.getenv('DB_URL')+":9200/dept/_search?size=1000&scroll=1m&pretty=true"
        response = requests.get(url)
        j = response.json()
        print(j, flush=True)
        a = []
        for hit in j["hits"]["hits"]:
            a.append( {"deptno": hit["_source"]["deptno"], "dname": hit["_source"]["dname"], "loc": hit["_source"]["loc"]} )
        response = jsonify(a)
        response.status_code = 200
        return response
    except Exception as e:
        print(traceback.format_exc(), flush=True)
        print(e, flush=True)

@app.route('/info')
def info():
        return "Python - Flask / OpenSearch" 

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=8080)

File: option/src/app/node_opensearch/src/rest.js
const express = require('express')
const app = express()
const https = require('https'); 
const port = 8080

app.get('/info', (req, res) => {
    res.send('NodeJS - Express / OpenSearch')
})

app.get('/dept', (req, res) => {

    var url = "https://"+process.env.DB_URL+":9200/dept/_search?size=1000&scroll=1m&pretty=true"
    console.log("url:" + url);

    https.get(url, function(http_res){
        var body = '';
        http_res.on('data', function(chunk){
            body += chunk;
        });
        http_res.on('end', function(){
            var j= JSON.parse(body);
            console.log(j);
            result = [];
            for (i in j.hits.hits) {
                hit = j.hits.hits[i]
                result.push({"deptno":hit._source.deptno,"dname":hit._source.dname,"loc":hit._source.loc })
            }
            res.send(result)
        });
    }).on('error', function(e){
          console.log("Got an error: ", e);
    });
})

app.listen(port, () => {
    console.log(`OCI Starter: listening on port ${port}`)
})
